bliss/simulator/decoder.py

```diff
@@ -124,54 +161,55 @@ def render_images(self, tile_cat: TileCatalog, image_ids):
             Tuple[Tensor, List, Tensor]: tensor of images, list of PSFs, tensor of PSF params
         """
         batch_size, n_tiles_h, n_tiles_w = tile_cat.n_sources.shape
-        assert len(image_ids) == batch_size
+        assert (
+            len(image_ids) == batch_size
+        ), "image_ids array should contain `batch_size` identical values"
 
         slen_h = tile_cat.tile_slen * n_tiles_h
         slen_w = tile_cat.tile_slen * n_tiles_w
-        images = np.zeros((batch_size, self.n_bands, slen_h, slen_w), dtype=np.float32)
+        image_shape = (self.n_bands, slen_h, slen_w)
+        images = np.zeros((batch_size, coadd_depth, *image_shape), dtype=np.float32)
 
         # use the PSF from specified image_id
         psfs = [self.psf_galsim[image_ids[b]] for b in range(batch_size)]
         param_list = [self.psf_params[image_ids[b]] for b in range(batch_size)]
         psf_params = torch.stack(param_list, dim=0)
 
         # use the specified nmgy_to_nelec ratios indexed by image_id
-        nmgy_to_nelec_rats = [self.nmgy_to_nelec_dict[image_ids[b]] for b in range(batch_size)]
+        nmgy_to_nelec_rats = [self.physical_to_nelec_dict[image_ids[b]] for b in range(batch_size)]
```

Comment:
better not to mention nanomaggies here

Reply:
because some surveys don't use these units

Reply:
Done.